hudi-spark-datasource/hudi-spark3.3.x/src/main/antlr4/org/apache/hudi/spark/sql/parser/HoodieSqlBase.g4

```diff
@@ -29,5 +29,12 @@ statement
     | createTableHeader ('(' colTypeList ')')? tableProvider?
         createTableClauses
         (AS? query)?                                                   #createTable
+    | CREATE INDEX (IF NOT EXISTS)? identifier ON TABLE?
```

Comment:
Could we still maintain the grammar in a single place for all Spark versions, but fail the logical plan of INDEX SQL statement in Spark 3.1 and below, so the grammar can be easily maintained?

Reply:
We cannot put the grammar in `HoodieSqlCommon.g4`, since we have to parse it in `HoodieSqlCommonAstBuilder` to build `CreateIndex`, while one param `FieldName` doesn't exist below Spark3.

We already have this for `time travel`, and grammar is rarely modified, so changing it like so is acceptable?

Reply:
Got it.  @codope is working on functional indexes in Hudi and may extend the grammar here.  So I'm thinking if it's not too hard to put the grammar in a common place, we should do that.